### ISSUE TITLE
'Latest release' info

### ISSUE THREAD

**Issue by keunes** [AntennaPod/antennapod.github.io#99]
**Short description**: Have a small block displaying the "Latest release" version name & release date and a "What's new" block displaying that version's 'What's new' info (as visible on Google Play and F-Droid). They should probably be visually connected (or a whole), and be updated automatically (without extra intervention for each new release).

**Location**: Homepage

**Why have this**: Show that AntennaPod is in active development; encourage visitors to download the app.

**More info**: 
* We could either [create a dynamic section](http://stackoverflow.com/a/26454035) or maybe (don't know if this is possible) from the app's (GitHub) Actions set something in motion that updates the static website?
* The 'What's new' section could either display the text capped, with a 'read more' button that expands the block (if this whole thing is presented further down on the page, where it can take more space), or just display the version & date with a 'View what's new' button/link (if this whole thing is displayed prominently above the fold, where we want to keep the view clean).
* displaying the latest release version (date + version number) and a 'what's new' block

**Comment by keunes** [comment]
Hi @nini1294 Sorry for replying only now. A long time ago I made a quick mock-up in GIMP for what the block could look like. However, I just made a proper [mock-up](https://design.penpot.app/#/view/d1c45d80-3b12-11ec-a516-23056af0be2f?page-id=d1c45d81-3b12-11ec-a516-23056af0be2f&index=0&share-id=84111db0-3b41-11ec-a516-23056af0be2f) for a more extensive homepage. I feel that it might not fit on the homepage, as it'd become too busy and it might not be relevant to people that get to the landing page.

But it would be great if it could be included on [the 'Download' page](https://antennapod.org/download/).

Here's the thing I drew - it's ugly but it indicates the elements I'd like to display:
![latest version](https://user-images.githubusercontent.com/11229646/139721887-d8fab4f7-6a4d-4efd-adf9-4c3b7db4077c.png)

Did you work on this in the meantime?

**Comment by nini1294** [comment]
Hey @keunes, that mockup looks good, it helps solidify the idea. I think the changelog box can be made modular so we can slot it into the page that makes the most sense.

I mostly got stuck with trying to get the relevant notes for a each release since some releases (like v2.3.1 and v2.3.2) are just bug fixes but it makes sense to get the detailed release notes (like for v2.3.0) even if the most recent release is smaller. The F-Droid release notes seem to always get them correctly, so I was trying to do something similar.

I'll probably try figuring that out this weekend hopefully.

**Comment by keunes** [comment]
This is great to hear!

> I mostly got stuck with trying to get the relevant notes for a each release since some releases (like v2.3.1 and v2.3.2) are just bug fixes but it makes sense to get the detailed release notes (like for v2.3.0) even if the most recent release is smaller. The F-Droid release notes seem to always get them correctly, so I was trying to do something similar.

That'd be great indeed. I don't know how that works in the background. If you don't manage, always having the latest release notes would already be nice :)

How/where do you intend to pull this information? Is it something that happens every time a user opens/refreshes the page, or would it be something that's included in a [cross-repo action](https://github.com/marketplace/actions/github-action-build-chain-cross-repo-builds) (where a site rebuild is triggered upon a new AP release)?

**Comment by keunes** [comment]
@nini1294 Just an update on where the F-Droid release notes come from, as @ByteHamster explained me just now: F-Droid and Google Play use the following file: https://github.com/AntennaPod/AntennaPod/blob/develop/app/src/main/play/release-notes/en-US/default.txt

This file gets updated manually for new releases, but it's often a few weeks ahead of the actual public release (for beta releases, etc). So if somehow it were possible to get the longer release notes from the _GitHub releases_, that would be better. Usually the first release (not tag) with a new second-level number (e.g. 1.**8**.0, 2.**0**.1, 2.**1**.1, 2.**2**.0) is the one that has the longer release notes which could be displayed also for subsequent (bug-fixing) releases. Maybe that could be a pattern that could be used.

**Comment by nini1294** [comment]
> How/where do you intend to pull this information? Is it something that happens every time a user opens/refreshes the page, or would it be something that's included in a cross-repo action (where a site rebuild is triggered upon a new AP release)?

I think this kind of automated action to cache them in the repo would be best, since pulling the notes from an API every time someone visits the page would cause too many requests. I can look into what the GitHub action syntax for that kind of hook would be.

That second URL you mentioned could be helpful since we can maybe look back at the version of that file at the time the release was tagged. For example the 2.3.2 version (https://github.com/AntennaPod/AntennaPod/blob/2.3.2/app/src/main/play/release-notes/en-US/default.txt) should have the relevant notes.

**Comment by keunes** [comment]
> That second URL you mentioned could be helpful since we can maybe look back at the version of that file at the time the release was tagged.

Wouldn't it be as easy but slightly more reliable to look back at the notes of a previous release, rather than a previous version of that text file? (Open question - I'm not a developer so I have no clue.)

**Comment by ByteHamster** [comment]
Should both work fine. I upload the release notes before tagging the release to have F-Droid show the correct ones. So just setting the tag number in the URL like above should return the correct release notes.

**Comment by nini1294** [comment]
Yep, it should definitely be possible, especially if that is being done in the GitHub action script itself rather than each time someone visits the page.

**Comment by keunes** [comment]
Sorry to bother you @nini1294 - I'm working on making the website easier to translate. I would like to avoid any potential code conflicts, plus I'd like to make sure the 'latest release' gets translated, too. I was wondering, therefore: would you have interest & time to work on this the coming week(s)? Then I can wrap up my stuff after :)

**Comment by nini1294** [comment]
Hey, sorry I've been swamped with other stuff, so haven't gotten a chance to work on this at all. Please don't wait on me to start working on what you are planning.

In case I get a chance, I can maybe start working on the Github action automation part, since that shouldn't conflict with any website pages.

**Comment by keunes** [comment]
Allright, cool. Thanks for replying!

(By the way, I'll need to do some work on GitHub Actions also for the translations - are you experienced with it? I might ask you to check my work because it's the first time I'm looking at it :) ).

**Comment by keunes** [comment]
Hi @nini1294,

Sorry to bother you again. Just to let you know that I wrapped up the major changes regarding website translations and also managed to implement some GitHub Actions for it. Would you still be interested in implementing the version indicator for the website?
Cheers

**Comment by keunes** [comment]
Hi @AnXh3L0,

I was wondering if you would be OK with opening a new PR for your code that introduces the 'latest release' button? (https://github.com/AntennaPod/antennapod.github.io/commit/8ecae38aea260e630124449e44d20d86f39ab7a3) Then (ideally) we can add commits to it to tweak it a bit - would be really cool to have this functionality!

**Comment by AnXh3L0** [comment]
Hey @keunes, sure, I will open a new PR for the latest release info.
